notes/lecture17.tex

```diff
@@ -0,0 +1,189 @@
+\begin{document}
+
+\section{Lecture 17: Non-Convex Problems}
+\sectionlabel{convexity}
+
+This lecture provides the important information on how non-convex problems differ from convex problems. The major issue in non-convex problems is that it can be difficult to find the global minimum because algorithms can easily get stuck in the possibly numerous local minima and saddle points.
+
+\begin{center}
+\includegraphics[width=1.0\linewidth, height=0.4\linewidth]{figures/lecture_17_non_convex_graph.png} 
+\end{center}
+
+\subsection{Local minima}
+
+\begin{definition}[Local minimum]
+A point $x^*$ is an unconstrained \textit{local minimum} if there exist $\epsilon > 0$ such that $f(x^*) \le f(x)$ for all $x$ with $\|x - x^*\| < \epsilon$.
+\end{definition}
+
+\begin{definition}[Global minimum]
+A point $x^*$ is an unconstrained \textit{global minimum} if $f(x^*) \le f(x)$ for all $x$.
+\end{definition}
+
+For both definitions, we say ''strict'' if these inequalities are strict.
+
+\begin{proposition}[Necessary Conditions for local minimum]
+\propositionlabel{nc for local min}
+Let $x^*$ be an unconstrained local minimum of $f\colon \R^n \to\R$ and assume $f$ is continuously differentiable ($C^1$) in an open set containing $x^*$. Then
+\begin{enumerate}
+    \item $\nabla f(x^*) = 0$ (First-Order Necessary Condition)
+    \item If in addition $f$ is twice continuously differentiable in an open set around $x^*$, then $\nabla^2f(x^*) \succeq 0$. (Second Order Necessary Condition)
+\end{enumerate}
+\end{proposition}
+
+\begin{proof}[Proof of \propositionref{nc for local min}.]
+Fix any direction $d \in \R^n$.
+\begin{enumerate}
+    \item $g(\alpha) := f(x^* + \alpha d)$. Then
+    \begin{align}
+        0 &\le \lim_{\alpha \to 0} \frac{f(x^* + \alpha d) - f(x^*)}{\alpha} \label{eqn:0} \\
+        &= \frac{\partial g(0)}{\partial \alpha} \nonumber \\
+        &= d^\trans \nabla f(x^*) \nonumber
+    \end{align}
+    \ref{eqn:0} Because $x^*$ is a local minimum, $0 \le f(x^* + \alpha d) - f(x^*)$ for sufficiently small alpha. So, we can construct a sequence with only positive $\alpha$ that converges to $x^*$ such that each element $0 \le \frac{f(x^* + \alpha_n d) - f(x^*)}{\alpha_n}$ which implies that statement given that $f$ is locally differentiable.\\
```

Comment:
Please don't start sentence with \ref{}. Say something like `Inequality~\ref{eqn:0} follows, because...`

To avoid conflict, please also prefix your labels with `lecture17`, e.g.  `\ref{lecture17-eqn0}`